### PR TITLE
docs: mention testing changes and keeping PRs up-to-date in PR docs

### LINKED ISSUE #8610: Docs: Explicitly say whether PR authors should merge from `main`
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

I've noticed that authors of PRs sometimes take on the task of keeping the PR updated from `main`. This happens...

* For PRs that haven't been reviewed yet and even for ones that have `1 approval` added
* For PRs with and without merge conflicts

My personal opinion is that we shouldn't expect PR authors to continuously merge the latest `main` into their branch. Especially as we've been more actively recently, it can be a real tax on some PRs to continuously merge from `main`. The worst case scenario would be that someone's PR takes a wihle to get reviewed -at no fault of theirs- and they keep feeling a need to resolve introduced merge conflicts in the interim. 😬 

And, as @auvred mentions below, subscribed folks get notifications for each merge.

Proposal: let's establish the recommended practice as...

* By default, no need to merge from `main`
    * If you see merge conflicts or other intersections that worry you, then you can as a courtesy
* If we think merge conflicts need to be resolved in order to merge and/or review a PR, we might opt into that as a courtesy _if_ we have time - but may still ask you to
    * If a PR spent an unusually long time to get a review, we'll try to find the time 

...where _"unusually long"_ and _"try to find the time"_ are intentionally vague.

~Edit: GitHub lost the data from the submission, re-typing after I walk away from the computer for a few moments...~ ✔️ 

### Affected URL(s)

https://typescript-eslint.io/contributing/pull-requests

### LINKED ISSUE #9210: Docs: Move PR-specific testing advice from Contributing > Local Development to Pull Requests
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

It's been bugging me for a while that I can never intuitively find our docs note to keep unit tests granular. I always go to [Contributing > Pull Requests](https://typescript-eslint.io/contributing/pull-requests) first, _then_ remember it's under [Contributing > Local Development > Validating Changes > Tests > Granular Unit Tests](https://typescript-eslint.io/contributing/local-development#granular-unit-tests).

I'd put it there in #7319 because the Pull Requests page didn't previously deal with PR contents, just the process around them. But in retrospect I don't agree with that decision 😄.

Along with and #8610 and #8997, as we flesh out the docs for PRs, I think we can reorganize to be a bit more friendly.

### Affected URL(s)

https://typescript-eslint.io/contributing/local-development
https://typescript-eslint.io/contributing/pull-requests

💖

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
## PR Checklist

- [x] Addresses an existing open issue: fixes #8610, fixes #9210
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken

## Overview

Lumps the two issues in together since they both touch the same page.

💖

### CODE REVIEW

File: docs/contributing/Pull_Requests.mdx [hunk 1]
```diff
@@ -84,6 +103,15 @@ Once you've addressed all our feedback by making code changes and/or started a f
...
 Once the feedback is addressed, and the PR is approved, we'll ensure the branch is up to date with `main`, and merge it for you.
 
+#### Updating Over Time
+
+You generally don't need to keep merging commits from `main` into your PR branch.
+If you see merge conflicts or other intersections that worry you, then you can preemptively - but that's optional.
+
+If we think merge conflicts need to be resolved in order to merge and/or review a PR, we might do that for you as a courtesy _if_ we have time.
+If not, we may ask you to.
+If a PR spent an unusually long time to get a review, we'll try our best to find the time to find the time.
```
[the remark this answers is not in the document]
Comment (JoshuaKGoldberg) [reply docs/contributing/Pull_Requests.mdx:113]: Oh I meant to convey that we'll try to help out. Clarifying.

File: docs/contributing/Pull_Requests.mdx [hunk 2]
```diff
@@ -84,6 +103,15 @@ Once you've addressed all our feedback by making code changes and/or started a f
...
 Once the feedback is addressed, and the PR is approved, we'll ensure the branch is up to date with `main`, and merge it for you.
 
+#### Updating Over Time
+
+You generally don't need to keep merging commits from `main` into your PR branch.
+If you see merge conflicts or other intersections that worry you, then you can preemptively - but that's optional.
+
+If we think merge conflicts need to be resolved in order to merge and/or review a PR, we might do that for you as a courtesy _if_ we have time.
+If not, we may ask you to.
+If a PR spent an unusually long time to get a review, we'll try our best to find the time to help with merging.
```
Comment (Josh-Cena) [inline docs/contributing/Pull_Requests.mdx:113]: I still don't see how it's relevant to this section, which talks about "when/if I should merge main into my branch". Here is says "we will do it for you, or we will ask you to"—sure. Is the last sentence supposed to mean "if we have zero activity, trust that we will get there one day, and we'll get it to a mergeable state ourselves without bothering you again"? This still feels more relevant about review frequency in the previous section. (Also I remember we have a sentence about how long one should wait before pinging a maintainer, maybe it fits there)

Comment (JoshuaKGoldberg) [reply docs/contributing/Pull_Requests.mdx:113]: Gotcha, makes sense. Thanks! I just removed the sentence.

**Review by Josh-Cena — APPROVED** [review]

File: docs/contributing/Pull_Requests.mdx [hunk 3]
```diff
@@ -84,6 +103,14 @@ Once you've addressed all our feedback by making code changes and/or started a f
 
 Once the feedback is addressed, and the PR is approved, we'll ensure the branch is up to date with `main`, and merge it for you.
 
+#### Updating Over Time
+
+You generally don't need to keep merging commits from `main` into your PR branch.
```
Comment (kirkwaiblinger) [inline docs/contributing/Pull_Requests.mdx:108]: You might mention if there's a preference for "merge" vs "rebase" if merging from main is done. 

I come from an environment where we'd always rebase against main and force push for this purpose, but it seems like merging from main is preferred in this codebase.

File: docs/contributing/Local_Development.mdx [hunk 4]
```diff
@@ -61,23 +61,6 @@ You can run `yarn test` in any package to run its tests.
...
-
-Most tests in most packages are set up as small, self-contained unit tests.
-We generally prefer that to keep tests and their failure reports granular.
-
-For rule tests we recommend, when reasonable:
-
-- Including both `valid` and `invalid` code changes in most PRs that affect rule behavior
-- Limiting to one error per `invalid` case
-
 ### Type Checking
```
Comment (kirkwaiblinger) [inline docs/contributing/Local_Development.mdx:64]: Hmm. should this section move too? 🤔

**Review by kirkwaiblinger — COMMENTED** [review]
github is requiring me to leave a top-level review comment in order to publish my individual review comments 🤦‍♂️

File: docs/contributing/Pull_Requests.mdx (hunk 3 above)
Comment (Josh-Cena) [reply docs/contributing/Pull_Requests.mdx:108]: This is already mentioned at the top of this page:

> Please don't:
>
> - Force push after opening a PR
>   - Reasoning: GitHub is not able to track changes across force pushes, which makes it take longer for us to perform incremental reviews

Comment (kirkwaiblinger) [reply docs/contributing/Pull_Requests.mdx:108]: Perhaps this is just a weird silo I'm in, but I work in an environment where

1. Thou shalt not create a merge commit
2. Though shalt not force push
3. 1. Takes precedence over 2. if there is a merge conflict. 

So, having read the contributing guide, I still, intentionally, force pushed the first time I had to resolve a merge conflict with main for a PR to typescript-eslint, thinking that was the right thing to do. 🤷‍♂️

Comment (Josh-Cena) [reply docs/contributing/Pull_Requests.mdx:108]: Hmmm, that does sound like a weird catch 22. I'm personally in the "always force push" camp. One sentence saying "merge not rebase" sounds good

File: docs/contributing/Local_Development.mdx (hunk 4 above)
Comment (JoshuaKGoldberg) [reply docs/contributing/Local_Development.mdx:64]: I don't think so - I see this page as the _"here are the raw commands you'll use"_.

**Pull request merged**